scripts/run_migration_between_software_builds.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash -e
+#
+# Generate migration scripts between current software deployment and older one 
+#
+
+# assumes current environment is current date tag
+CUR_DATE_TAG=$(module --redirect show spack | grep ".lua" | sed "s:/: :g" | awk '{print $3}')
+CUR_SPACK_VERSION=$(module --redirect show spack | grep ".lua" | sed "s:/: :g" | awk '{print $NF}' | sed "s/.lua://g")
+
+if [ -z "${OLD_DATE_TAG}" ]; then
+    OLD_DATE_TAG="2022.11"
+fi
+
+echo "Producing migration scripts in $(pwd) directory"
+echo "Scripts are spack.specs.sh and spack.install.sh for checking the specs and for installing them"
+echo "Migration is for ${MYPROJECT} project and ${MYSOFTWARE} for ${OLD_DATE_TAG} software stack deployment"
+ 
+# unload the default gcc compiler and swap to old pawsey environment to load old spack module
+module unload gcc 
+module swap pawseyenv pawseyenv/${OLD_DATE_TAG}
+module load gcc
+OLD_SPACK_VERSION=$(module --redirect show spack | grep ".lua" | sed "s:/: :g" | awk '{print $NF}' | sed "s/.lua://g")
+module load spack/${OLD_SPACK_VERSION}
+
+# look for modules within your $MYSOFTWARE installation
+hashlist=($(lfs find /software/projects/pawsey0001/pelahi/setonix/modules/ -name *.lua | sed "s/.lua//g" | sed "s/-/ /g" | awk '{print $2}'))
```

Comment:
As deva pointed out, we should use `${PAWSEY_PROJECT}` and `${USER}`.

Reply:
Yup, I did but must have accidentally copied this line from the working  one that I was testing. Will fix.